### PR TITLE
Add AKS extension and flux configuration modules

### PR DESCRIPTION
**Pull request by rahalan**
# Change

Add AKS extension and flux configuration modules

[KubernetesConfiguration: Extensions](https://github.com/Azure/ResourceModules/actions/workflows/ms.kubernetesconfiguration.extensions.yml)

[KubernetesConfiguration: FluxConfigurations](https://github.com/Azure/ResourceModules/actions/workflows/ms.kubernetesconfiguration.fluxconfigurations.yml)

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update (Wiki)

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (readme)
- [x] I did format my code

### CODE REVIEW

File: arm/Microsoft.KubernetesConfiguration/extensions/deploy.bicep [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  name: name
+  scope: managedCluster
+  properties: {
+    autoUpgradeMinorVersion: autoUpgradeMinorVersion
+    configurationProtectedSettings: !empty(configurationProtectedSettings) ? configurationProtectedSettings : {}
+    configurationSettings: !empty(configurationSettings) ? configurationSettings : {}
+    extensionType: extensionType
+    releaseTrain: !empty(releaseTrain) ? releaseTrain : null
+    scope: {
+      cluster: empty(releaseNamespace) ? null : {
```
Comment (MrMCake) [inline arm/Microsoft.KubernetesConfiguration/extensions/deploy.bicep:63]: Please flip the condition (same for the other cases)

File: arm/Microsoft.KubernetesConfiguration/extensions/readme.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+```powershell
+az provider register --namespace Microsoft.Kubernetes
+az provider register --namespace Microsoft.ContainerService
+az provider register --namespace Microsoft.KubernetesConfiguration
+```
+
+For Details see [Prerequisites](https://docs.microsoft.com/en-us/azure/azure-arc/kubernetes/tutorial-use-gitops-flux2)
+
+## Navigation
```
Comment (MrMCake) [inline arm/Microsoft.KubernetesConfiguration/extensions/readme.md:23]: Move Navigation to the top.

It's also a bit odd that it did not automatically add it as a header... Can you re-run the script after the move?

File: arm/Microsoft.KubernetesConfiguration/fluxConfigurations/readme.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+```powershell
+az provider register --namespace Microsoft.Kubernetes
+az provider register --namespace Microsoft.ContainerService
+az provider register --namespace Microsoft.KubernetesConfiguration
+```
+
+For Details see [Prerequisites](https://docs.microsoft.com/en-us/azure/azure-arc/kubernetes/tutorial-use-gitops-flux2)
+
+## Navigation
```
Comment (MrMCake) [inline arm/Microsoft.KubernetesConfiguration/fluxConfigurations/readme.md:23]: Some comment as before - please move and try to generate in the hopes the table is correctly updated

**Comment by rahalan** [comment]
> 

Done

File: arm/Microsoft.KubernetesConfiguration/extensions/deploy.bicep (hunk 1 above)
Comment (rahalan) [reply arm/Microsoft.KubernetesConfiguration/extensions/deploy.bicep:63]: done

File: arm/Microsoft.KubernetesConfiguration/extensions/readme.md (hunk 2 above)
Comment (rahalan) [reply arm/Microsoft.KubernetesConfiguration/extensions/readme.md:23]: done

File: arm/Microsoft.KubernetesConfiguration/fluxConfigurations/readme.md (hunk 3 above)
Comment (rahalan) [reply arm/Microsoft.KubernetesConfiguration/fluxConfigurations/readme.md]: done

**Review by MrMCake — APPROVED** [review]

**Pull request merged**